Optimalbb PR for review purpose

sloriot opened the pull request.

maxGimeno committed:
Make collapse_edge always keep the target vertex.

maxGimeno committed:
Make Isotropic_remeshing keep the constrained vertices when collapsing an edge.

maxGimeno committed:
Fix after rebase

maxGimeno committed:
updates changes.md

maxGimeno committed:
Update doc for edge_collapse

sloriot committed:
the corner status of a vertex is already taken into account in the code

it is meaningful for collapse and smooth that are using is_corner function

sloriot committed:
fix indentation

sloriot committed:
a vertex of degree less than 3 is on the boundary

collapsing an edge incident to it should be controlled only
by the constrained status of the vertices

lrineau committed:
WIP: implementation for scalar image

... tested only on segmented images!

lrineau committed:
Protection of triple lines of the bbox for scalar images

Now that work. I have to work a bit on the precision of the approximation.

sloriot committed:
always preserve q at collapse by swapping p and q if q is of degree 2

the advantage is that we have an easy way to document which
halfedges are removed

sloriot committed:
update documentation

sloriot committed:
simplify the code collapsing edges

handles corner and constrained vertices

sloriot committed:
update changes

sloriot committed:
consider any vertex incident to a constrained as constrained

sloriot committed:
swap halfedges to guarantee that constrained halfedges are kept

this is only requires in case prev(opposite(he)) is constrained
since if there is a constrained vertex it is vb.

lrineau committed:
Fix a bug

and add assertions. The assertions use lambda expression to display useful
information.

maxGimeno committed:
WIP testsuite collapse_edge

maxGimeno committed:
test halfedges in test_Collapse_edge.cpp

maxGimeno committed:
Change float to double in the arguments of find_halfedge

maxGimeno committed:
add comments on cases and remove code from asserts

lrineau committed:
Fix a line of code that was never tested

lrineau committed:
Add debug instructions

lrineau committed:
Rename the file

lrineau committed:
Move Index_generator to another header

lrineau committed:
WIP: allow to change the Surface_patch_index type

lrineau committed:
Use CGAL::cpp11::function to store functors

and at the same time the constructors logic is factorized in the
constructor of a base class.

lrineau committed:
WIP: Change the template parameters of Labeled_image_mesh_domain_3 (TODO)

TODO: document the changes, once that is over.

lrineau committed:
Add Construct_surface_patch_index to the Impl_details

lrineau committed:
Call iso_cuboid in the Impl_details base class

lrineau committed:
Replace Default by the real type default

lrineau committed:
Cleanup included header for Boost Parameters

lrineau committed:
Add a constructor using Boost Parameters

lrineau committed:
Set BOOST_PARAMETER_MAX_ARITY very early

lrineau committed:
Fix -Wconversion warnings

lrineau committed:
Correct usage of Boost Parameters

lrineau committed:
Remove -Wconversion warnings

lrineau committed:
Cleaning of impl. details in <CGAL/Mesh_3/Image_to_labeled_function_wrapper.h>

Now that wrapper class:
 - uses `std::function`,
 - is not longer templated by the kernel,
 - or by the image type.

lrineau committed:
Rename a function for readability

lrineau committed:
Use <boost/noncopyable.hpp> instead of our home-made solution

lrineau committed:
Missing #include

lrineau committed:
Remove that annoying warning

When a user-project is compile from a Git repository, there was that
annoying warning about the project not being developed in the same branch
as CGAL itself!

lrineau committed:
Fix a -Wconversion warning

lrineau committed:
Correct use of CGAL_NO_DEPRECATED_CODE

lrineau committed:
Add Labeled_mesh_domain_3<K>::create_gray_image_mesh_domain

... and deprecate `CGAL::Gray_image_mesh_domain_3`.

lrineau committed:
Remove the -Wdeprecated-declarations warning

lrineau committed:
Use the new API

lrineau committed:
Fix the code when Surface_patch_index is not std::pair<int,int>

lrineau committed:
Use the new API of Labeled_mesh_domain_3 for the demo

lrineau committed:
Add Labeled_mesh_domain_3<K>::create_labeled_image_mesh_domain

... and deprecate `CGAL::Labeled_image_mesh_domain_3`.

lrineau committed:
Fix -Wconversion warnings

lrineau committed:
Use boost/shared_ptr.hpp to re-enable move semantic

lrineau committed:
Remove the domain wrapper file, that is no longer used

lrineau committed:
Add an example to mesh multiples isolevels of a 3D gray image

lrineau committed:
Handle the image offset

lrineau committed:
result_type is used by polylines_to_protect()

lrineau committed:
Fix compilation in C++03

lrineau committed:
Support image offsets in polylines_to_protect

lrineau committed:
Force nodes on the edges of the cube to be terminal nodes

lrineau committed:
fixup! Force nodes on the edges of the cube to be terminal nodes

lrineau committed:
Remove an unused template parameter

lrineau committed:
Document the new API of Labeled_mesh_domain

lrineau committed:
Document the deprecation of the old domain classes from 3D images

lrineau committed:
Fix formatting

From Mael comment:
https://github.com/CGAL/cgal/pull/2739#discussion_r175404665

lrineau committed:
Add the C++ concept Callable

Fix Mael's comment:
https://github.com/CGAL/cgal/pull/2739#discussion_r175413586

lrineau committed:
Document where the geometric objects come from (from BGT)

See Mael's comment:
https://github.com/CGAL/cgal/pull/2739#discussion_r175414307

lrineau committed:
Unify sentence "Construct a... " and "Construction from"

See Mael's comment:
https://github.com/CGAL/cgal/pull/2739#discussion_r175405016

lrineau committed:
New wording

See Mael's comment:
https://github.com/CGAL/cgal/pull/2739#discussion_r175410322

lrineau committed:
Wording

See Mael's comment:
https://github.com/CGAL/cgal/pull/2739#discussion_r175448829

lrineau committed:
Typo: missing )

lrineau committed:
Remove trivial, useless, buggy documentation comments

lrineau committed:
Remove the domain classes deprecated since CGAL-4.5.

Mael talked about that in his review.

lrineau committed:
Deprecate Implicit_mesh_domain_3 as well

lrineau committed:
Adapt the test suite to the deprecation of Implicit_mesh_domain_3

lrineau committed:
Remove the last -Wconversion warnings from all Mesh_3

I have only compiled tests and examples. Maybe the Mesh_3 plugin of the 3D
demo has other warnings.

lrineau committed:
Remove the buggy concept LabelImage_3

lrineau committed:
Remove mentions of the deprecated classes

lrineau committed:
Document CGAL::Image_3

lrineau committed:
typo: missing space

lrineau committed:
Document create_implicit_mesh_domain()

lrineau committed:
Remove the `const bool = true` from the function wrappers

lrineau committed:
Add deprecation notice to Implicit_mesh_domain_3, in the doc

lrineau committed:
Remove unused typedef

lrineau committed:
Add (deprecated) in the list of classes

MaelRL committed:
Reworked Fuzzy_sphere

Same as a6f52ffc31029bb1d3e2c075413766c4a14ddfd9

- Also fixed the way eps > r is handled
- Cached the inner and outer squared radius instead of recomputing every time

MaelRL committed:
Improved tests of Fuzzy sphere and iso box

MaelRL committed:
Fixed boost include

MaelRL committed:
Fixed typo

MaelRL committed:
Improved tests: always add random points on inner/outer approximation boundaries

lrineau committed:
Fix warnings

lrineau committed:
Fix a compilation bug if Mesh_3 and Mesh_2 are both used

The conflict is withing the Boost parameters defined for Lloyd 2D and
Lloyd 3D.

lrineau committed:
Use static_evaluate

lrineau committed:
Fix compilation bug with CGAL_MESH_3_PROTECTION_DEBUG

lrineau committed:
Do not store const& to the function/functor by copy by value

However, one cannot store a function type directly, but only *pointers* to
function types. So I used Boost MPL to discriminate between function types
and other callable (such as lambdas or functors, that can be copied).

MaelRL committed:
Use CGAL::square

MaelRL committed:
Readability changes

MaelRL committed:
Do not use an inner approx if 'eps > r'

MaelRL committed:
Improved class documentation of fuzzy models

MaelRL committed:
Fixed warning

MaelRL committed:
'if' into 'else if' since min_coord(i) <= max_coord(i)

kkatrio committed:
qr factorization

kkatrio committed:
fitness function

kkatrio committed:
simplex functions refection, contraction, etc

kkatrio committed:
centroid function

kkatrio committed:
population class

lrineau committed:
Document that the implicit function is CopyConstructible

kkatrio committed:
nelder-mead algorithm

kkatrio committed:
genetic algorithm

kkatrio committed:
visualization

maxGimeno committed:
Add a property_map that converts points from a cartesian kernel to another, for a mesh.

maxGimeno committed:
Clean-up

kkatrio committed:
it works!

maxGimeno committed:
Fixes

maxGimeno committed:
Forward declare the Cartesian_converter to avoid messing with the dependencies tree.

maxGimeno committed:
reset dependencies

kkatrio committed:
tests

kkatrio committed:
polyhedron demo plugin

MaelRL committed:
Fixed warning (?)

My kingdom for readable Windows compilation messages

MaelRL committed:
Fixed constructing field type from non-int variables

MaelRL committed:
Properly propagate the random seed

kkatrio committed:
cleaning

kkatrio committed:
use convex hull

kkatrio committed:
selection of faces on the plugin

kkatrio committed:
stopping criteria on generations

kkatrio committed:
cleaning

sloriot committed:
make it work with any Kernel object

kkatrio committed:
demo plugin: works on point set data

sloriot committed:
fix the handling of constrained vertices

sloriot committed:
restore non-wanted change

sloriot committed:
fix doc

kkatrio committed:
cleaning

afabri committed:
VC++ has a problem accessing matrix entries with operator()

afabri committed:
Do not do a matrix-matrix multiplication, but do N matrix-vector multiplications as we only want min and max

maxGimeno committed:
Update CHANGES.md

kkatrio committed:
cleaning

maxGimeno committed:
Clean-up after review

kkatrio committed:
benchmarks

maxGimeno committed:
Fix doc

kkatrio committed:
abstract interface for eigen dense matrix

sloriot committed:
try improving phrasing

sloriot commented on Optimal_bounding_box/include/CGAL/Optimal_bounding_box/fitness_function.h:
```diff
@@ -0,0 +1,183 @@
+    return compute_fitness(R, point_data);
+  }
+
+  double compute_fitness(Matrix3d& R, MatrixXd& data)
```
const

sloriot commented on Optimal_bounding_box/include/CGAL/Optimal_bounding_box/fitness_function.h:
```diff
@@ -0,0 +1,183 @@
+    double xmin, xmax, ymin, ymax, zmin, zmax;
+    for(int i = 0; i < data.rows(); i++){
+
+      Vector3d vec = data.row(i);
```
Linear_algebra_traits::row(data, i);

kkatrio committed:
eigen linear algebra traits

sloriot committed:
improve phrasing

lrineau committed:
Merge branch 'master' into Property_map-Point_kernel_converter_map-GF

lrineau committed:
Fix compatibility with C++03

lrineau committed:
Add the license-checking #include

lrineau committed:
Merge branch 'master' into Mesh_3-improve_polylines_to_protect-GF

kkatrio committed:
clear warnings & clean eigen eigen linear algrebra traits

kkatrio committed:
cleaning

kkatrio committed:
API linear algebra parameter

lrineau committed:
Fix check_headers

kkatrio committed:
making sure that all matrix allocations happen at compile time when possible

kkatrio committed:
draw perfomance graph

kkatrio committed:
benchmark code and benchmark on fitness function

kkatrio committed:
small errors

kkatrio committed:
I hate c++

kkatrio committed:
return precompiled 3-column row only & some cleaning

kkatrio committed:
use a matrix with dynamic rows and columns are run time for points, because it is a little faster than fixed columns

kkatrio committed:
documentation

lrineau committed:
Fix CDT_2 errors, using snapping of intersection points

When `Constrained_triangulation_2` computes the intersection of two
segments, with a floating-point number type, and with
`Exact_predicates_tag`, the computed intersection point is snapped to an
extremity of the two segments, if it is closest to 4 ulp (with the l-inf
distance).

That value `4` can be changed by defining the macro
`CGAL_CDT_2_INTERSECTION_SNAPPING_ULP_DISTANCE` to another value.

afabri committed:
fix off file so that OpenMesh can read it too

kkatrio committed:
python script setting up path

kkatrio committed:
remove mutable and fix small typo

Commit:
add missing const to compare_endpoints_xy_2 () operator taking x-monotone-curve as a parameter

afabri committed:
fix include guards

afabri committed:
Use CGAL::abs

afabri committed:
Don't copy the vector

afabri committed:
one less size_t/int warning

afabri commented on Optimal_bounding_box/include/CGAL/Optimal_bounding_box/obb.h:
```diff
@@ -0,0 +1,253 @@
+// works on matrices only
+/// \cond SKIP_IN_MANUAL
+template <typename Vertex, typename Matrix>
+void post_processing(const Matrix& points, Vertex& R, Matrix& obb)
```
Parameterize with the LinearAlgebraTraits which should get  a `typedef ..  Index`, which is `int` for Eigen. This is then the type that should be used instead of a hard code `std::size_t` in this function.

afabri committed:
Move typedef to the right scope

afabri commented on Optimal_bounding_box/include/CGAL/Optimal_bounding_box/nelder_mead_functions.h:
```diff
@@ -0,0 +1,99 @@
+namespace Optimal_bounding_box {
+
+#if defined(CGAL_EIGEN3_ENABLED)
+typedef CGAL::Eigen_linear_algebra_traits Linear_algebra_traits;
```
You cannot simply place this `typedef` in the namespace.  @sloriot do you agree?

afabri commented on Optimal_bounding_box/include/CGAL/Optimal_bounding_box/population.h:
```diff
@@ -0,0 +1,152 @@
+{
+
+#if defined(CGAL_EIGEN3_ENABLED)
+  typedef CGAL::Eigen_linear_algebra_traits Linear_algebra_traits;
```
Population should be templated with LinearAlgebraTraits and this traits class then gives you  the Matrix type.

afabri committed:
Move #include

afabri commented on Optimal_bounding_box/include/CGAL/Optimal_bounding_box/obb.h:
```diff
@@ -0,0 +1,253 @@
+  // get the ch3
+  if(use_ch)
+  {
+    // find the ch - todo: template kernel
```
Let's remove the dependency on the convex hull from obb.h and the API. 

Add a function to the package Convex_hull_3 that given a range of points writes the convex hull points into an output iterator.

afabri commented on Optimal_bounding_box/include/CGAL/Optimal_bounding_box/obb.h:
```diff
@@ -0,0 +1,253 @@
+    std::vector<Kernel::Point_3> ch_points(poly.points_begin(), poly.points_end());
+
+    // points: vector -> matrix
+    CGAL::Optimal_bounding_box::fill_matrix(ch_points, points_mat);
```
I am pretty sure that this will not wok for `Exact_predicates_exact_construction_kernel::Point_3`. You must use `CGAL::to_double()`

afabri committed:
typo

afabri commented on Optimal_bounding_box/include/CGAL/Optimal_bounding_box/obb.h:
```diff
@@ -0,0 +1,253 @@
+template <typename Point, typename LinearAlgebraTraits>
+void find_obb(std::vector<Point>& points,
+              std::vector<Point>& obb_points,
+              LinearAlgebraTraits&,
```
There should be a default for the linear algebra.

afabri commented on Optimal_bounding_box/include/CGAL/Optimal_bounding_box/obb.h:
```diff
@@ -0,0 +1,253 @@
+/// @param use_ch a bool flag to indicating whether to use the convex hull of the input points
+/// as an optimization step.
+template <typename Point, typename LinearAlgebraTraits>
+void find_obb(std::vector<Point>& points,
```
When we had discussed the API, we wanted a function that takes a range of points  and returns the transformation matrix. @sloriot did you have a `CGAL::Affine_transformation_3` in mind or a `LinearAlgebraTraits::Matrix3d` ?

afabri committed:
Add Bbox_{2,3}.dilate(int) so that the snapping in the CT_2 also works with Projection_traits classes

kkatrio committed:
function overload for convex_hull_3 which returns a list of points instead of a polyhedron

afabri committed:
Fix doc

kkatrio committed:
requested changes: impoving API with default Linear algebra traits, better use of template parameters at free functions, prefer using convex hull overload which does not depend on the kernel.

maxGimeno committed:
return if less that 2 selected items

afabri committed:
fix Bbox_2::dilate()

MaelRL committed:
Fixed taking a reference to a temporary (when using EPECK)

MaelRL committed:
Improved RT3's tests of "insertion with info" types of functions

maxGimeno committed:
Add missing updates after the fork of libQGLViewer

lrineau committed:
Fix a Markdown syntax error in the CHANGES.md

lrineau committed:
Merge 'master' and fix conflicts

maxGimeno committed:
Add the check headers in the CHECK pass of travis

MaelRL committed:
Fixed random seed propagation (bis)

MaelRL committed:
Added guard object 'Set_ieee_double_precision'

Fixes issues on 32-bit platforms, see:
https://github.com/CGAL/cgal/pull/2954#issuecomment-384974067

maxGimeno committed:
Update cmake files

sloriot committed:
move code swapping vertices in an internal helper function

maxGimeno committed:
Add dependency to selection_plugin

lrineau committed:
Workaround for MSVC

The MSVC is confused by the name `Function` being at the same time a
template parameter and a typedef of the base class.

```
...\include\CGAL/Implicit_mesh_domain_3.h(73): error C2248: 'CGAL::Labeled_mesh_domain_3<BGT,int,std::pair<int,int>>::Function': cannot access private typedef declared in class 'CGAL::Labeled_mesh_domain_3<BGT,int,std::pair<int,int>>'
        with
        [
            BGT=K
        ]
```

lrineau committed:
Merge branch 'master' into Property_map-Point_kernel_converter_map-GF

lrineau committed:
Attempt to fix the compilation issue with MSCV 2012 and Boost 1.53

```
[ 50%] Building CXX object CMakeFiles/mesh_implicit_sphere_variable_size.dir/mesh_implicit_sphere_variable_size.cpp.obj
	C:\PROGRA~2\MICROS~1.0\VC\bin\amd64\cl.exe  @C:\Users\afabri\AppData\Local\Temp\nmCD61.tmp
mesh_implicit_sphere_variable_size.cpp
C:\3rdPartyLibs\boost_1_53_0\boost/type_traits/remove_cv.hpp(37) : error C2039: 'unqualified_type' : is not a member of 'boost::detail::cv_traits_imp<T>'
        with
        [
            T=double (__cdecl *const *)(const Point &)
        ]
```
https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-4.13-Ic-23/Mesh_3_Examples/TestReport_afabri_x64_Cygwin-Windows8_MSVC2012-Release-64bits.gz

Maybe passing a function pointer instead of a function lvalue will fix the
compilation error.

afabri committed:
Replace M_PI with CGAL_PI

sloriot committed:
use a directory that exists

MaelRL committed:
Fixed optimization plugin not using the new domain API

MaelRL committed:
Fixed missing includes

sloriot committed:
move code to swap edges in a function

sloriot committed:
make the function collapsing edges with constraints always keep the target

MaelRL committed:
Fixed including deprecated header

afabri committed:
An #ifdef inside a CGAL_assertion_msg() does not compile on VC12

maxGimeno committed:
Fix call to nproc

kkatrio committed:
handle convex hull functionality for up to 3 points to a mutable graph and to an output iterator

sloriot committed:
fix mathjax default for doxygen 1.8.13

lrineau committed:
Do not include the headers of deprecated classes

maxGimeno committed:
Try to fix opencv bug with EXIT

lrineau committed:
Add a test of CGAL_MESH_3_VERBOSE

lrineau committed:
Add a comment to retrieve the issue quickly

lrineau committed:
Also add a test for mesher.one_step()

lrineau committed:
Fix the CGAL_MESH_3_VERBOSE mode

lrineau committed:
partially revert commit 89d96f4ce8f

lrineau committed:
Fix of CHANGES.md

The faulty modification was made by the commit:

    commit 851dfabd297ec89b224a9d02c4f13a7767d92716
    Merge: 46bf950f2d6 d3c874bd83d
    Author: Laurent Rineau <laurent.rineau@cgal.org>
    Date:   Mon May 28 13:58:07 2018 +0200

        Merge pull request #2757 from maxGimeno/BGL-Add_facegraph_validity_check_function-GF

        BGL:  Add a  FaceGraph validity check function

lrineau committed:
Merge pull request #3122 from MaelRL/T23-Fix_functor_return_type-GF

Triangulation_23: fix wrong return type in spatial_sort functor

lrineau committed:
Merge pull request #3098 from lrineau/Triangulation_2-Debug_CDT2-lrineau

Fix CDT_2 errors, using snapping of intersection points

lrineau committed:
Merge pull request #2954 from MaelRL/Spatial_searching-Fix_fuziness_doc-GF

Spatial searching: rework fuzzy items

lrineau committed:
Merge pull request #3122 from MaelRL/T23-Fix_functor_return_type-GF

Triangulation_23: fix wrong return type in spatial_sort functor

# Conflicts:
#	Triangulation_3/test/Triangulation_3/test_regular_insert_range_with_info.cpp

lrineau committed:
Merge pull request #3131 from sloriot/Doc-update_dir

Use a directory that exists

lrineau committed:
Merge pull request #3098 from lrineau/Triangulation_2-Debug_CDT2-lrineau

Fix CDT_2 errors, using snapping of intersection points

lrineau committed:
Merge pull request #2954 from MaelRL/Spatial_searching-Fix_fuziness_doc-GF

Spatial searching: rework fuzzy items

# Conflicts:
#	Spatial_searching/test/Spatial_searching/Point_with_info.h

lrineau committed:
Merge pull request #3135 from sloriot/Doc-mathjax_1.8.13

Fix default mathjax URL for doxygen 1.8.13

lrineau committed:
Merge branch 'releases/CGAL-4.11-branch' into releases/CGAL-4.12-branch

lrineau committed:
Merge pull request #2954 from MaelRL/Spatial_searching-Fix_fuziness_doc-GF

Spatial searching: rework fuzzy items

# Conflicts:
#	Spatial_searching/test/Spatial_searching/Point_with_info.h

lrineau committed:
Merge pull request #3118 from maxGimeno/Demo-Fix_Set_different_colors-GF

Polyhedron demo: Fix for Set Different Colors for Selected Items

lrineau committed:
Merge pull request #3122 from MaelRL/T23-Fix_functor_return_type-GF

Triangulation_23: fix wrong return type in spatial_sort functor

# Conflicts:
#	Triangulation_3/test/Triangulation_3/test_regular_insert_range_with_info.cpp

lrineau committed:
Merge pull request #3123 from maxGimeno/Manual_update_after_QGLViewer_integration-GF

Documentation: Add missing updates after the fork of libQGLViewer

lrineau committed:
Merge pull request #2739 from lrineau/Mesh_3-improve_polylines_to_protect-GF

Mesh_3 new improvements: polylines_to_protect(), new API of Labeled_mesh_domain_3

lrineau committed:
Merge pull request #3130 from afabri/CGAL-M_PI-GF

Replace M_PI with CGAL_PI

lrineau committed:
Merge pull request #3131 from sloriot/Doc-update_dir

Use a directory that exists

lrineau committed:
Merge pull request #3098 from lrineau/Triangulation_2-Debug_CDT2-lrineau

Fix CDT_2 errors, using snapping of intersection points

lrineau committed:
Merge pull request #3135 from sloriot/Doc-mathjax_1.8.13

Fix default mathjax URL for doxygen 1.8.13

lrineau committed:
Merge branch 'releases/CGAL-4.12-branch'

MaelRL committed:
std::unary_function > CGAL::unary_function

afabri committed:
Add a typedef Point Point_ to be used in a method definition

maxGimeno committed:
Fix warning.

maxGimeno committed:
Add missing includes in other plugins

afabri committed:
Undo adding the _. Instead remove the typedef ... Point;

afabri committed:
Replace Point by Point_3 in Surface_mesher demo

lrineau committed:
Merge pull request #3139 from lrineau/Mesh_3-fix_verbose_mode-lrineau

Mesh_3: fix verbose mode (when the manifold criterion is used)

lrineau committed:
Merge pull request #3109 from psarahdactyl/master

add missing const to compare_endpoints_xy_2 () operator taking x-mono…

lrineau committed:
Merge pull request #3143 from MaelRL/T3-Fix_sneaky_std_unary_function_bis_repetita-GF

Triangulation_3: Fix std::unary_function leftover

lrineau committed:
Merge pull request #3150 from afabri/Triangulation_3-Conflict_with_Windows_Point-GF

Polyhedron demo: Remove conflicting 'Point' typedef

lrineau committed:
Merge pull request #3139 from lrineau/Mesh_3-fix_verbose_mode-lrineau

Mesh_3: fix verbose mode (when the manifold criterion is used)

lrineau committed:
Merge pull request #3124 from maxGimeno/Travis-check_headers-GF

Travis: Add  check_headers in the CHECK pass of travis

lrineau committed:
Merge pull request #3143 from MaelRL/T3-Fix_sneaky_std_unary_function_bis_repetita-GF

Triangulation_3: Fix std::unary_function leftover

lrineau committed:
Merge pull request #2737 from maxGimeno/BGL_isotropic_remeshing_keep_constrained_points_intact-GF

BGL: Isotropic_remeshing() keeps constrained vertices intact

# Conflicts:
#	Installation/CHANGES.md

lrineau committed:
Merge pull request #3058 from maxGimeno/Property_map-Point_kernel_converter_map-GF

Property_map: Kernel_converter_map

lrineau committed:
Merge pull request #3124 from maxGimeno/Travis-check_headers-GF

Travis: Add  check_headers in the CHECK pass of travis

lrineau committed:
Merge pull request #3127 from maxGimeno/Demo-Fix_selection_io_plugin-GF

Polyhedron demo: Fix Selection_io_plugin

lrineau committed:
Merge branch 'releases/CGAL-4.12-branch'

lrineau committed:
Cherry-pick pull request #3109 from psarahdactyl/master

add missing const to compare_endpoints_xy_2 () operator taking x-monotone-curve as a parameter

lrineau committed:
Merge branch 'releases/CGAL-4.12-branch'

sloriot committed:
clean up

kkatrio committed:
added forgotten const qualifiers
